test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceMappingProviderTests.cs

```diff
@@ -46,6 +46,30 @@ public void GetPackageSourceMappingItems_WithOneConfig_ReturnsCorrectPatterns()
             packageSourceMappingSourceItem.Patterns.First().Pattern.Should().Be("stuff");
         }
 
+        [Fact]
+        public void GetPackageSourceMappingItems_WithOneConfig_WithDuplicateKeys_Throws()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSourceMapping>
```

Comment:
I believe you need `</clear>` here, because you're getting `Package source 'True' has already been defined previously` in CI build. Probably need to investigate what is that `True` source.

Reply:
I had an error in the code, I was returning if the key existed or not, I updated the code

Reply:
Tons of tests are failing. I'll review again after tests are fixed.